src/test/java/com/example/music_player/service/SourceServiceTest.java

```diff
@@ -51,105 +57,82 @@ public void setup() {
         testArray = "testByteArray".getBytes();
         testBoolean = false;
         sourceList = List.of(source1, source2);
+        inputStream = new ByteArrayInputStream(StandardCharsets.UTF_16.encode("testString").array());
     }
 
     @Test
-    void save() {
-
+    void saveWhenSourceDoNotExist() throws IOException {
+        when(sourceRepository.isExistByNameAndFileType(song.getName()
+                , mockMultipartFile.getContentType()))
+                .thenReturn(testBoolean);
+        when(storageSourceService.save(
+                mockMultipartFile.getInputStream()
+                , mockMultipartFile.getName()
+                , mockMultipartFile.getContentType()))
+                .thenReturn(sourceList);
+        doNothing().when(sourceRepository).save(source1);
+
+        sourceService.save(mockMultipartFile, song, song.getId());
+        assertEquals(testBoolean, false);
+        verify(storageSourceService, times(1))
+                .save(mockMultipartFile.getInputStream()
+                        , mockMultipartFile.getName()
+                        , mockMultipartFile.getContentType());
+        verify(sourceRepository, times(1)).save(source1);
```

Comment:
You forgot to verify that sourceRepository is invoked for all the sources from sourceList. To not forget verify all invocations you can add Mockito.verifyNoMoreInteractions(allYourMocks) at the end of all test scenarios.